src/useBaseAxios.ts

```diff
@@ -21,14 +24,17 @@ function useBaseAxios<Data>(param1: string | AxiosRequestConfig, param2: AxiosRe
 
   const invokeAxios =
     typeof param1 === 'string'
-      ? () => axios(param1, { ...param2, cancelToken })
-      : () => axios({ ...param1, cancelToken });
+      ? (lazyData: AxiosRequestConfig['data']) =>
+          axios(param1, { ...param2, data: lazyData || param2.data, cancelToken })
+      : (lazyData: AxiosRequestConfig['data']) => {
```

Comment:
You shouldn't need to pass in `lazyData` here since it's [already in scope](https://github.com/angelle-sw/use-axios-client/pull/43/files#diff-f5dc9592405f21d5dae8a80a1e7a986dR13).